README.md

```diff
@@ -45,11 +45,48 @@ exercise club noble adult miracle awkward problem olympic puppy private goddess
 ```console
 $ cardano-address key from-recovery-phrase Shelley < phrase.prv > root.xsk
 root_xsk1hqzfzrgskgnpwskxxrv5khs7ess82ecy8za9l5ef7e0afd2849p3zryje8chk39nxtva0sww5me3pzkej4rvd5cae3q3v8eu7556n6pdrp4fdu8nsglynpmcppxxvfdyzdz5gfq3fefjepxhvqspmuyvmvqg8983
+
+-- which is equivalent to
+
+$ cardano-address key from-recovery-phrase Shelley --passphrase "" < phrase.prv
+root_xsk1hqzfzrgskgnpwskxxrv5khs7ess82ecy8za9l5ef7e0afd2849p3zryje8chk39nxtva0sww5me3pzkej4rvd5cae3q3v8eu7556n6pdrp4fdu8nsglynpmcppxxvfdyzdz5gfq3fefjepxhvqspmuyvmvqg8983
 ```
 
 > :information_source: Notice the `root_xsk` prefix to identify a root extended signing (private) key.
 </details>
 
+<details>
+  <summary>How to generate a root private key with passphrase (<strong>root.xsk</strong>)</summary>
+
+```console
+$ cardano-address recovery-phrase generate --size 9 > sndfactor.prv
+swing payment diagram happy chimney mammal flip become lyrics
+
+$ cardano-address key from-recovery-phrase Shelley --passphrase "$(cat sndfactor.prv)" < phrase.prv
```

Comment:
You generally want to avoid passing sensitive stuff like passphrases as command-line options, as these remains in the shell history afterwards. I would suggest to offer two alternatives to users:

- Provide a passphrase via a file; 
- Or else, if no file is provided (but only `--passphrase`), enter the passphrase interactively.

Reply:
see the added section in Readme.md to see all supported options

Reply:
So currently we have several options:
```
cardano-address key from-recovery-phrase Shared --passphrase from-mnemonic
cardano-address key from-recovery-phrase Shared --passphrase from-hex
cardano-address key from-recovery-phrase Shared --passphrase from-base64
cardano-address key from-recovery-phrase Shared --passphrase from-utf8
cardano-address key from-recovery-phrase Shared --passphrase from-octets
```
Each of those options triggers interactive mode accordingly, e.g.:
```
Please enter a [9, 12, 15, 18, 21, 24] word mnemonic:
skill consider antenna lunar arch coach coast symbol rent
Please enter passphrase in the form of octet array:
[220,56]
```
As far as I see there are two issues:

1. The interactive mode does not allow making any mistake... when user types wrong thing he's not able to correct it as hitting Backspace, "arrow" or del adds additional characters:
```
Please enter a [9, 12, 15, 18, 21, 24] word mnemonic:
ere ere ^[[D^[[D^?^?^[[3~^[[3~^[[C^[[C
```

2. Currently there's no non-interactive mode that would allow user to set `--passphrase` without going into interactive mode. I believe that non-interactive mode would be quite useful if one would like to use --passphrase feature in some automation workflow:

I think that what @KtorZ proposes makes sense:
>Provide a passphrase via a file;

which would enable non-interactive mode, e.g.:
```
cardano-address recovery-phrase generate \
| cardano-address key from-recovery-phrase Shared --passphrase /path/to/phrase.prv
```

> Or else, if no file is provided (but only --passphrase), enter the passphrase interactively (with confirmation)

Which should be good calling it like this:
```
cardano-address recovery-phrase generate \
| cardano-address key from-recovery-phrase Shared --passphrase
Please enter additional passphrase in the form of utf-8:
```
or like this:
```
cardano-address key from-recovery-phrase Shared --passphrase
Please enter a [9, 12, 15, 18, 21, 24] word mnemonic:
...
Please enter additional passphrase in the form of utf-8:
...
```

I also think that from practical sense having options: from-mnemonic, from-hex, from-base64, from-octet is not particularly useful. It seems that plain utf-8 should be enough for additional `--passphrase`.

Reply:
well, `from-mnemonic` is definitely useful as we want seamless compatibility with cardano-wallet which uses mnemonic as second factor in api. The rest are optional. One needs to remember that not always mnemonic passphrase maps to utf8 passphrase (but aways to base16 or base64 and of course octet). Nevertheless, I see the point of adding possibility to read from file :+1:

Reply:
@piotr-iohk So now we have the following:
1. we have backspace support in interactive mode
2. we have explicit, sensitive and silent mode 
3. we have support for all from-format modes in file mode
4. docs updated
5. added comprehensive testing for from-file